extensions/vertx/deployment/src/main/java/io/quarkus/vertx/core/deployment/VertxCoreProcessor.java

```diff
@@ -286,6 +286,12 @@ ContextHandlerBuildItem createVertxContextHandlers(VertxCoreRecorder recorder, V
         return new ContextHandlerBuildItem(recorder.executionContextHandler(buildConfig.customizeArcContext()));
     }
 
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public void cleanUp(VertxCoreRecorder recorder, ShutdownContextBuildItem shutdown) {
+        recorder.resetMapper(shutdown);
+    }
+
```

Comment:
This is causing issues in native mode and since it's only meant to fix a dev-mode case, we should limit it's applicability to that mode